src/history/hs_rec.c

```diff
@@ -183,6 +201,8 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
     WT_STAT_DATA_INCR(session, cache_hs_insert);
 
 err:
+    if (!hs_read_all_flag)
```

Comment:
If the flag was set originally, should we set it again here?
If we reach here without error, the flag should have been cleared?

Reply:
We can't be sure if we had to reach here due to a WT_ERR() statement and not going through the code that can restore flags.

Reply:
What do you mean? We should restore the flag no matter how we reached here I think.

Reply:
If we have done so before, we still can safely do this and restore the flag. If you only want to restore in failure. You can check the ret value.

Reply:
Yes that's correct. Best case, we reached here without an error and we have already restored the flags. But in case of an error we might never had a chance to restore the flags because we jumped to error handling.

Reply:
I see you changed the previous code. I think it is OK now.